Add docs on Airflow 3 compatibility (#1731)
astronomer/astronomer-cosmos: docs/airflow3_compatibility/index.rst

Airflow 3 Compatibility (First Iteration)
=========================================

The Cosmos 1.10.0 release marks the **first iteration** of adding compatibility for `Apache Airflow® 3 <https://airflow.apache.org/>`_
This is an important milestone as we work towards ensuring that Cosmos seamlessly integrates with the latest advancements in the Airflow ecosystem.

> What's Been Tested

Reviewer: ```suggestion
What Works
```

------------------

With the changes contributed in Cosmos 1.10.0 to bring in compatibility with Airflow 3, we have validated Cosmos’
functionality with Airflow 3 by extending our CI infrastructure:

- **All example DAGs** in the `dev/dags <https://github.com/astronomer/astronomer-cosmos/tree/main/dev/dags>`_ directory of the repository are tested against Airflow 3.
- Our `CI test matrix <https://github.com/astronomer/astronomer-cosmos/blob/main/.github/workflows/test.yml>`_ now includes explicit entries to run the following tests on Airflow 3:
  - Unit Tests
  - Integration Tests
  - Performance Tests
  - Kubernetes Tests

These additions ensure that all core functionality, workflows, and integrations provided by Cosmos continue to operate
reliably under Airflow 3.

> Experimental Features

Reviewer: ```suggestion
Validation in Progress
```

---------------------

We are actively validating the combined support for `Assets <https://airflow.apache.org/docs/apache-airflow/3.0.0/authoring-and-scheduling/assets.html>`_
and `OpenLineage <https://airflow.apache.org/docs/apache-airflow-providers-openlineage/stable/guides/user.html>`_ with Airflow 3.

> While early results are promising, we are marking this integration as **experimental** in Cosmos 1.10.0.

Reviewer: ```suggestion
This may be **unstable** in Cosmos 1.10.0. Bug reports are very welcome.
```
